contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,208 @@
-    // @param markets the markets to claim fees from
-    // @param tokens the fee tokens to claim
-    function claimFees(
-        address[] memory markets,
-        address[] memory tokens
-    ) external nonReentrant onlyFeeKeeper {
-        if (markets.length != tokens.length) {
-            revert Errors.InvalidClaimFeesInput(markets.length, tokens.length);
+    // @dev withdraw fees in buybackTokens from this contract
+    // @param marketTokens the markets from which to withdraw fees
+    // @param buybackToken the token for which to withdraw fees
+    function withdrawFees(address[] calldata markets, address buybackToken) external nonReentrant onlyFeeKeeper {
+        uint256 batchSize = dataStore.getUint(Keys.buybackBatchAmountKey(buybackToken));
+        _validateBuybackToken(batchSize, buybackToken);
+
+        for (uint256 i; i < markets.length; i++) {
+            FeeUtils.claimFees(dataStore, eventEmitter, markets[i], buybackToken, address(this));
         }
 
         address receiver = dataStore.getAddress(Keys.FEE_RECEIVER);
+        IERC20(buybackToken).safeTransfer(receiver, IERC20(buybackToken).balanceOf(address(this)));
+    }
+
+    // @dev claim fees in feeToken from the specified markets
+    // @param market the market from which to claim fees
+    // @param feeToken the fee tokens to claim from the market
+    function claimFees(address market, address feeToken) external nonReentrant {
+        uint256 batchSize = dataStore.getUint(Keys.buybackBatchAmountKey(feeToken));
+        if (batchSize != 0) {
+            revert Errors.InvalidFeeToken(feeToken);
+        }
+
+        uint256 feeAmount;
+        if (market == address(0)) {
+            address gov = IVaultV1(vaultV1).gov();
+            feeAmount = IVaultGovV1(gov).withdrawFees(vaultV1, feeToken, address(this));
+            _setAvailableFeeAmounts(1, feeToken, feeAmount);
+        } else {
+            feeAmount = FeeUtils.claimFees(dataStore, eventEmitter, market, feeToken, address(this));
+            _setAvailableFeeAmounts(2, feeToken, feeAmount);
+        }
+    }
+
+    // @dev receive an amount in feeToken by depositing the batchSize amount of the buybackToken
+    // @param feeToken the token to receive with the fee amount calculated via an oracle price
+    // @param buybackToken the token to deposit in the amount of batchSize in return for fees
+    // @param minOutputAmount the minimum amount of the feeToken that the caller will receive
+    function buyback(address feeToken, address buybackToken, uint256 minOutputAmount) external nonReentrant {
+        if (feeToken == buybackToken) {
+            revert Errors.BuybackAndFeeTokenAreEqual(feeToken, buybackToken);
+        }
+
+        uint256 batchSize = dataStore.getUint(Keys.buybackBatchAmountKey(buybackToken));
+        _validateBuybackToken(batchSize, buybackToken);
+
+        uint256 availableFeeAmount = _getAvailableFeeAmount(feeToken, buybackToken);
```

Comment:
may be good to revert if availableFeeAmount is zero

Reply:
Ok, no problem. I added another AvailableFeeAmountIsZero error that returns (address feeToken, address buybackToken, uint256 availableFeeAmount) since it didn't seem to make sense to return minOutputAmount as is now done in the InsufficientBuybackOutputAmount error.